code/go/0chain.net/smartcontract/storagesc/blobber.go

```diff
@@ -685,7 +685,13 @@ func (sc *StorageSmartContract) commitBlobberConnection(
 	}
 
 	// the first time the allocation is added  to the blobber, created related resources
-	if blobAlloc.BlobberAllocationsPartitionLoc == nil {
+	if blobAlloc.Stats.UsedSize == 0 {
```

Comment:
Isn't the UsedSize incremental only? so this is actually checking if the blobber is redeeming a read marker that its allocation had no data been written to it yet, is that possible?

Reply:
@cnlangzi please confirm that if the WriteMarker.Size could be negative.

Reply:
@sum-it08 @peterlimg  WriteMaker.Size can be any value( >0,0,<0). eg
- upload/update: >0
- delete: <=0
- rename: 0